src/views/ProfilePage.jsx

```diff
@@ -1,8 +1,29 @@
 import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { login, logout } from '../store/userSlice'
+
 
 function ProfilePage() {
+
+  const user = useSelector((state) => state.user.user)
+  const dispatch = useDispatch()
+
   return (
-    <div>Profile page</div>
+    <div>Profile page
+      <button
+          onClick={() => dispatch(login())}
```

Comment:
on line 10 in the `userSlice` you are setting the user to be equal to the payload of the login action
`state.user = action.payload`
however here the payload you are passing is `null` (you are not passing one ^^)
it needs to be
`() => dispatch(login('Lena'))`
;)

Reply:
Oh, now it works! :)) Thanks!

Reply:
in general for these onClick parameters I recommend writing the handler function outside the JSX code. This is because the logic can end up getting complex, so it makes the component harder to read.

something like this: 

```
function ProfilePage() {
   const user = useSelector((state) => state.user.user)
   const dispatch = useDispatch()
   
   function handleLogin() {
      dispatch(login('Lena'))
   }
   
   function handleLogout() {
      dispatch(login('Lena'))
   }
   
   return (
     <div>
     ...
        <button onClick={handleLogin}>Login</button>
     ...
     </div>
   )
}